### PR TITLE
Lstm

### PR DESCRIPTION
**Pull request by Pierre28**

### CODE REVIEW

File: src/classifier.py [hunk 1]
```diff
@@ -1,30 +1,87 @@
...
+import numpy as np
+
+import spacy
+
 from collections import Counter
 import re
-import torch.optim as optim
+import os 
+from time import time
 torch.manual_seed(1)
```
Comment (Marsobad) [inline src/classifier.py]: Woow, this is a long file with so many classes! Maybe you could split this into multiple scripts. I know, the preprocessing step depends on the model but I think it would be more "modular" if there were a module for each step of the pipeline. For example, there could be one module for preprocessing sentences and everybody could call methods from it that will be different or similar depending on the model. 
An example of arborescence : 
- classifier.py : classifier + encoder
- preprocessing.py : SentencePreprocessor + Dataset

This script is also very long to read, it would be easier to understand it if each task was split in my opinion.

File: src/tester.py [hunk 2]
```diff
@@ -40,10 +42,10 @@ def eval_list(glabels, slabels):
 def train_and_eval(classifier, trainfile, devfile, testfile, run_id):
     print(f"\nRUN: {run_id}")
     print("  %s.1. Training the classifier..." % str(run_id))
-    classifier.train(trainfile)
```
Comment (Marsobad) [inline src/tester.py]: The professor said it was very important to keep the tester.py file unmodified. 
Maybe you just changed it during the test, but before merging it into master, I think it is relevant to remove arguments that did not belong to the original file and to change them directly in classifier.py.

File: requirement.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+spacy
+pytorch
```
Comment (Marsobad) [inline requirement.txt]: change to torch

**Review by Marsobad — CHANGES_REQUESTED** [review]
Great job @Pierre28  ! It works very well on my computer and accuracy is good. Just a few comments : 
- Could you split classifier.py into multiple scripts doing different tasks? I think it's very long and monolithic right now and it would be more readable and modular if each task was separate. Like there could be one preprocessing_tools.py file and a classifier.py file for example. 
- There's a typo in requirement
- Could you leave Tester.py unmodified please? It's written in red in the project description :p 
Otherwise, that's awesome, thank you very much , you rock 👍

**Review by Marsobad — APPROVED** [review]
Great
